src/tests/testHtmlQuote.cc

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/cppunit.h"
+#include "html_quote.h"
+#include "unitTestMain.h"
+
+#include <cstring>
+#include <iostream>
+
+class testHtmlQuote: public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(testHtmlQuote);
+    CPPUNIT_TEST(test_html_quote_cstr);
+    // CPPUNIT_TEST(testPerformance);
+    CPPUNIT_TEST_SUITE_END();
+
+    protected:
+    void test_html_quote_cstr();
+    void testPerformance();
+};
+
+CPPUNIT_TEST_SUITE_REGISTRATION( testHtmlQuote );
+
+void
+testHtmlQuote::test_html_quote_cstr()
+{
+    CPPUNIT_ASSERT_EQUAL(std::string(""), std::string(html_quote("")));
+    CPPUNIT_ASSERT_EQUAL(std::string("bar"), std::string(html_quote("bar")));
+    CPPUNIT_ASSERT_EQUAL(std::string("foo&lt;bar&gt;gazonk"), std::string(html_quote("foo<bar>gazonk")));
+    CPPUNIT_ASSERT_EQUAL(std::string("foo&amp;bar"), std::string(html_quote("foo&bar")));
+    CPPUNIT_ASSERT_EQUAL(std::string("some&#39;thing"), std::string(html_quote("some'thing")));
+    CPPUNIT_ASSERT_EQUAL(std::string("some&quot;thing"), std::string(html_quote("some\"thing")));
+    CPPUNIT_ASSERT_EQUAL(std::string("&#31;"), std::string(html_quote("\x1f")));
+}
```

Comment:
Pease add a loop that encodes each and every possible byte value. You would not be able to fully validate the encoding result (without heroic efforts), but please do check that the output length is positive and,

* if the output length is one, then the output character is the input character;
* if the first output character is `&`, then the last character is `;`;
* if the first output character is `&` and the second character is `#`, then output length is at least 4 (`&#1;`) and at most 6 (`&#255;`).

I do not insist on adding the check in the last/third bullet point above.

Reply:
Github copilot did most of the legwork :)
More than that, it exposes that this function takes signed chars, but the actual type that would make sense is unsigned